### PR TITLE
ENH: Add Python wrapping for SpatialOrientationAdapter

### PR DESCRIPTION
**Pull request by dzenanz**
To simplify this, SpatialOrientationAdapter no longer
inherits from OrientationAdapterBase.

Also add test to exercise Python wrapping of SpatialOrientationAdapter.
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [x] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

### CODE REVIEW

File: Documentation/ITK5MigrationGuide.md [hunk 1]
```diff
@@ -452,6 +452,9 @@ The namespece itself is only available in legacy mode. Some classes were renamed
 * `itk::SpatialOrientation::CoordinateMajornessTerms` became `itk::SpatialOrientationEnums::CoordinateMajornessTerms`.
 * `itk::SpatialOrientation::ValidCoordinateOrientationFlags` became `itk::SpatialOrientationEnums::ValidCoordinateOrientations`.
 
+Since ITK 5.3, `OrientationAdapterBase` is deprecated. It will be available in `ITKDeprecated` module.
```
Comment (thewtex) [inline Documentation/ITK5MigrationGuide.md:455]: We should not be moving classes into different modules until a major release version.

Comment (dzenanz) [reply Documentation/ITK5MigrationGuide.md]: Force-push changes that.

**Comment by dzenanz** [comment]
The first force-push makes a small change, the second is a plain rebase.

**Review by thewtex — APPROVED** [review]
:+1: nice!

**Pull request merged**